boards/launchxl/Makefile

```diff
@@ -0,0 +1,27 @@
+# Makefile for building the tock kernel for the launchxl development kit
+
+TOCK_ARCH=cortex-m4
+TOCK_BOARD=launchxl
+TARGET=thumbv7em-none-eabi
+PLATFORM=launchxl
+
+DSLITE=$(UNIFLASH_CLI_BASE)/ccs_base/DebugServer/bin/DSLite
+PROG_BASE=0x47000
+
+include ../Makefile.common
+
+apps/%/build/$(TOCK_ARCH)/$(TOCK_ARCH).bin:
+	@make -C apps/$*
+
+flash: guard-UNIFLASH_CLI_BASE target/$(TARGET)/release/launchxl
+	@echo "Flashing target/$(TARGET)/release/launchxl"
+	@$(DSLITE) flash -c jtag/cc2652r1f.ccxml -e -f target/$(TARGET)/release/launchxl
+
+flash-%: apps/%/build/$(TOCK_ARCH)/$(TOCK_ARCH).bin guard-DSLITE
+	@$(DSLITE) flash -c jtag/cc2652r1f.ccxml -e -f $<,$(PROG_BASE)
+
+guard-%:
+	@ if [ "${${*}}" = "" ]; then \
+		echo "Environment variable $* not set, see readme for more details."; \
+		exit 1; \
+	fi
```

Comment:
I think I'd prefer to see the `check_defined` from the userland make system be replicated here for consistency. You'll get errors like this then:
`Makefile:15: *** Undefined DSLITE (See launchxl README for details).  Stop.`
Here's a patch that does that:

```diff
diff --git a/boards/Makefile.common b/boards/Makefile.common
index 326d971f..f938b8bd 100644
--- a/boards/Makefile.common
+++ b/boards/Makefile.common
@@ -20,9 +20,23 @@ OBJCOPY   ?= $(TOOLCHAIN)-objcopy
 OBJDUMP   ?= $(TOOLCHAIN)-objdump
 OBJDUMP_FLAGS += --disassemble-all --source --disassembler-options=force-thumb -C --section-headers

-ifndef PLATFORM
-$(error Required make vairable PLATFORM is not defined)
-endif
+
+# http://stackoverflow.com/questions/10858261/abort-makefile-if-variable-not-set
+# Check that given variables are set and all have non-empty values,
+# die with an error otherwise.
+#
+# Params:
+#   1. Variable name(s) to test.
+#   2. (optional) Error message to print.
+check_defined = \
+    $(strip $(foreach 1,$1, \
+        $(call __check_defined,$1,$(strip $(value 2)))))
+__check_defined = \
+    $(if $(value $1),, \
+      $(error Undefined $1$(if $2, ($2))))
+
+
+$(call check_defined, PLATFORM)

 # If environment variable V is non-empty, be verbose
 ifneq ($(V),)
diff --git a/boards/launchxl/Makefile b/boards/launchxl/Makefile
index f345ad85..bc753346 100644
--- a/boards/launchxl/Makefile
+++ b/boards/launchxl/Makefile
@@ -10,18 +10,15 @@ PROG_BASE=0x47000

 include ../Makefile.common

+$(call check_defined, DSLITE, See launchxl README for details)
+$(call check_defined, UNIFLASH_CLI_BASE, See launchxl README for details)
+
 apps/%/build/$(TOCK_ARCH)/$(TOCK_ARCH).bin:
        @make -C apps/$*

-flash: guard-UNIFLASH_CLI_BASE target/$(TARGET)/release/launchxl
+flash: target/$(TARGET)/release/launchxl
        @echo "Flashing target/$(TARGET)/release/launchxl"
        @$(DSLITE) flash -c jtag/cc2652r1f.ccxml -e -f target/$(TARGET)/release/launchxl

-flash-%: apps/%/build/$(TOCK_ARCH)/$(TOCK_ARCH).bin guard-DSLITE
+flash-%: apps/%/build/$(TOCK_ARCH)/$(TOCK_ARCH).bin
        @$(DSLITE) flash -c jtag/cc2652r1f.ccxml -e -f $<,$(PROG_BASE)
-
-guard-%:
-       @ if [ "${${*}}" = "" ]; then \
-               echo "Environment variable $* not set, see readme for more details."; \
-               exit 1; \
-       fi
```

You can also put the check_defined calls in a target's build formula if they should only trigger for some targets (as the guards were doing before)

Reply:
Awesome, I'll update it!